tests/clickhouse-test

```diff
@@ -752,7 +750,7 @@ class SettingsRandomizer:
         "prefer_localhost_replica": lambda: random.randint(0, 1),
         "max_block_size": lambda: random.randint(8000, 100000),
         "max_joined_block_size_rows": lambda: random.randint(8000, 100000),
-        "max_threads": lambda: random.randint(1, 64),
+        "max_threads": lambda: random.randint(1, 3),
```

Comment:
This feels funny, because it's detecting a situation where we are scheduling more threads than we need even if they are not necessary. Still it seems ok.

Reply:
I've added some randomization of these settings